### ISSUE TITLE
Use inotify (pyinotify) to watch file system access/changes

### ISSUE THREAD

**Issue by SleepyMan** [z411/trackma#57]
Instead of periodically checking if a player is open using lsof, it will be more efficient to just listen to file system events through the kernel.
It's possible to watch the player binary (which mpv) and listen for open events (actually running it) and close_nowrite events (closing it).
If desired, it can be used to watch the media folder for new files, thus we would only need to run the file name regex/transform only once per file, instead of every time we press play next or something like that (which i believe it's once per minute, as I get a "can't stat fuse.sshfs" warnings once I'm not on the same network as my search folder)

This is a small enhancement, but it would remove the periodical check, and make it possible to watch those less than 1 minute web anime (excluding opening/ending sequences) without making it check for the player every 30 seconds (even if setting the tracker_update_wait to 0).

Edit: This would impose a new requirement. The kernel needs inotify and pyinotify support which is only available to ≥ 2.6.13. Though I don't think this is a problem.

Edit2: For reference: https://github.com/seb-m/pyinotify

**Comment by z411** [comment]
Hello, I implemented inotify in the "inotify" branch, if you have time please check it out.
Indeed Trackma now reacts immediately to the player, even faster than Windows alternatives; it's pretty cool.

**Comment by z411** [comment]
Hello, I implemented inotify in the "inotify" branch, if you have time please check it out.
Since I didn't need all the features of pyinotify, I decided for a lighter library called "inotifyx". You can install it through pip or through your distro repos. It falls back to polling if it's not found.

Indeed Trackma now reacts immediately to the player, even faster than Windows alternatives; it's pretty cool. Thanks for the suggestion.

**Comment by z411** [comment]
Merged into master. Added optional inotifyx dependency; falls back to polling if not found.

**Comment by veluria** [comment]
I've tested this and it only works for me if the file is in the root media directory. Subfolders do not seem to work. 

Also, this poses the problem that files which are played from everywhere else will not be recognized. I sometimes download files, just leave them in my Download folder and delete them afterwards. As such, I'd at least propose the addition of the possibility to add multiple media directories.

**Comment by z411** [comment]
Yes, this is normal as inotify is designed to watch specific directories for changes. I'll consider adding multiple directories, thank you.

**Comment by z411** [comment]
Moved that suggestion to a new issue (#89)
